cirq/circuits/circuit.py

```diff
@@ -1370,21 +1371,24 @@ def __radd__(self, other):
         result._device.validate_circuit(result)
         return result
 
-    def __imul__(self, repetitions: int):
-        if not isinstance(repetitions, int):
+    # Needed for numpy to handle multiplication by np.int64 correctly.
+    __array_priority__ = 10000
+
+    def __imul__(self, repetitions: INT_TYPE):
+        if not isinstance(repetitions, (int, np.int32, np.int64)):
             return NotImplemented
-        self._moments *= repetitions
+        self._moments *= int(repetitions)
         return self
 
-    def __mul__(self, repetitions: int):
-        if not isinstance(repetitions, int):
+    def __mul__(self, repetitions: INT_TYPE):
+        if not isinstance(repetitions, (int, np.int32, np.int64)):
```

Comment:
Consider including other numpy integer types to avoid surprises (e.g. int8, int16).

Reply:
relevant: https://stackoverflow.com/a/37727662

Reply:
Thanks for the link.  That's super useful.  I have added a bunch of different types to the test and changed the check to np.integer.